src/api_rework.jl

```diff
@@ -0,0 +1,120 @@
+# I am putting everything in here for now. The contents of this file should be moved around in the future.
+"""
+    findpeaks(x) -> NamedTuple
+
+Find the peaks in a vector `x`. 
+A `NamedTuple` is returned with the original vector 
+in the field `data`, and the indices of the peaks 
+in the field `inds`.
+
+This function serves as the entry-point for other 
+functions such as `peakproms!` and `peakwidths!`
+"""
+function findpeaks(x::AbstractVector)
```

Comment:
I've held off on a function named `findpeaks` in anticipation of someday adding different peak finding algorithms (e.g. wavelet based, etc). I haven't yet figured out what the API for that needs to look like to have a transparent default that could be extended in the future. So for now, lets skip this function name.

Reply:
Oh snap, changing the peak finding function name to `findpeaks` was one of my favorite changes. It just feels like the most plain alternative, and it aligns greatly with the package name. It is also what you will find in [python (scipy)](https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.find_peaks.html) and [MATLAB](https://www.mathworks.com/help/signal/ref/findpeaks.html), so users coming from other languages will find another name unnatural.

The peak-finding function that returns a named tuple will neccesarily have a different name from the original `findmaxima` unless we fiddle with positional argument (type stable) or keyword arguments. In any case the argument-controls-output-type approach would have to default to the old API to avoid breakage, meaning that the new API would be opt-in, i.e. we would guide users away from it. Not a good solution as I see it.

Then for your reason to skip this function name. If the algorithms do the same, I would (as a user) much prefer to have a consistent function name, and change the algorithm with an argument. This is conventional for most frameworks, no? [Optim.jl](https://julianlsolvers.github.io/Optim.jl/stable/#user/minimization/) uses the third positional argument, [SciML](https://docs.sciml.ai/DiffEqDocs/stable/getting_started/) uses the second argument, and I can not think of a package that uses seperate functions for performing the same operation with a different algorithm.

So if we want to introduce a new function that returns a named tuple, I would _love_ it to be named `findpeaks`. This avoids breakage, and introduces the new function name that aligns better with other languages and the package name.

Reply:
> ... have a consistent function name, and change the algorithm with an argument

Yes that is what I mean/intend; I just haven't spent the time to investigate what the most common/accepted Julian idiom is, and I see this function as orthogonal to this PR (which I'd like to narrow the focus to the new NamedTuple API).

Reply:
Given that I still feel like we need seperate API's that do not cross over, I still think we need a new entry point, and that `findpeaks` is clearly both common and accepted given usage in other languages. Another example is [FindPeaks](https://github.com/tungli/Findpeaks.jl) which uses `findpeaks`. There is also [FindPeaks1D](https://github.com/ymtoo/FindPeaks1D.jl), which uses `findpeaks1d`. I am actually hard-pressed to find examples where `findpeaks` is _not_ the name given to a function that, well, finds peaks.